components/kyma-metrics-collector/pkg/process/process.go

```diff
@@ -353,25 +353,75 @@ func isSuccess(status int) bool {
 	return false
 }
 
-func isClusterTrackable(runtime *kebruntime.RuntimeDTO) bool {
-	if runtime.Status.Provisioning != nil &&
-		runtime.Status.Provisioning.State == string(kebruntime.StateSucceeded) &&
-		runtime.Status.Deprovisioning == nil {
+func (p *Process) isClusterTrackable(runtime *kebruntime.RuntimeDTO) bool {
+	// Check if the cluster is in a trackable state
+	trackableStates := map[kebruntime.State]bool{
```

Comment:
(nit) How about we create a method like:
```
func isTrackableState(state string) bool {
	switch state {
	case "succeeded", "error", "upgrading", "updating":
		return true
	}
	return false
}
```

Reply:
We can even use `string(kebruntime.StateXYZ)` instead of strings.

Reply:
good point, done